### PR TITLE
Support ETS-GRNCH dataset

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated (with summary of main changes)
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds support for cleaning up the ETS-GRNCH dataset
* Adds functionality for returning size of nested lists within a dict
* Changes controlled data license from `private` to `restricted`
* Revised list of supported snow variables in ERA5/-Land

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

File: miranda/convert/_data_corrections.py [hunk 1]
```diff
@@ -67,6 +69,39 @@ def _iter_vars_key(ds, meta, key, project):
...
+) -> xr.Dataset:
+    filename = d.encoding["source"]
+    date = date_parser(Path(filename).name.split(split)[location])
+    entries = len(d[time_field])
+    date_range = pd.date_range(date, periods=entries, freq=freq)
+
+    d[time_field] = xr.DataArray(
+        time_field,
+        date_range,
+        attrs=dict(units=f"days since {date}", calendar="standard"),
```
Comment (aulemahal) [inline miranda/convert/_data_corrections.py:87]: If the coordinate has a datetime dtype, the `units` and  `calendar` attributes _should not_ be set, they are implicit in the dtype.

File: miranda/convert/_data_corrections.py [hunk 2]
```diff
@@ -426,9 +461,12 @@ def metadata_conversion(d: xr.Dataset, p: str, m: Dict) -> xr.Dataset:
     return d
 
 
-def variable_conversion(ds: xr.Dataset, project: str) -> xr.Dataset:
+def variable_conversion(
+    ds: xr.Dataset, project: str, files: Optional[List[str]] = None
```
Comment (aulemahal) [inline miranda/convert/_data_corrections.py:465]: What is the purpose of `files` here?

File: miranda/convert/data/ets_grnch_attrs.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      "cell_methods": "area: mean time: mean",
+      "history": "",
+      "long_name": "precipitation",
+      "standard_name": "precipitation_flux",
+      "units": "kg m-2 s-1"
+    },
+    "T": {
+      "_cf_variable_name": "tas",
+      "_corrected_units": "degC",
+      "_transformation": "amount2rate",
```
Comment (aulemahal) [inline miranda/convert/data/ets_grnch_attrs.json:51]: Temperature rates?

File: miranda/convert/data/ets_grnch_attrs.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      "standard_name": "air_temperature",
+      "units": "K"
+    },
+    "Time": {
+      "_cf_variable_name": "time",
+      "axis": "T",
+      "calendar": "standard",
+      "long_name": "time",
+      "standard_name": "time",
+      "units": "days since 1970-01-01"
```
Comment (aulemahal) [inline miranda/convert/data/ets_grnch_attrs.json:65]: As said above, you don't need to set the `units` and `calendar` if the data has a datetime dtype.

File: miranda/convert/data/ets_grnch_attrs.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      "history": "",
+      "long_name": "Daily Maximum Near-Surface Air Temperature",
+      "standard_name": "air_temperature",
+      "units": "K"
+    },
+    "Tmin": {
+      "_cf_variable_name": "tasmin",
+      "_corrected_units": "degC",
+      "_transformation": "amount2rate",
+      "cell_measures": "area: areacella",
```
Comment (aulemahal) [inline miranda/convert/data/ets_grnch_attrs.json:82]: Does the `areacella` variable exist in the dataset? While the CF conventions allows this attribute even if the variable is not there, I find it a bit strange.

File: miranda/convert/_data_corrections.py (hunk 2 above)
Comment (Zeitsperre) [reply miranda/convert/_data_corrections.py:465]: It was an idea I abandoned to address an issue with data, but the problem demands fixes on open, so not appropriate at this step. removed.

File: miranda/convert/data/ets_grnch_attrs.json (hunk 5 above)
Comment (Zeitsperre) [reply miranda/convert/data/ets_grnch_attrs.json:82]: It does not. Remove?

File: miranda/convert/data/ets_grnch_attrs.json (hunk 3 above)
Comment (Zeitsperre) [reply miranda/convert/data/ets_grnch_attrs.json]: woops. Removed.

File: miranda/convert/data/ets_grnch_attrs.json (hunk 5 above)
Comment (aulemahal) [reply miranda/convert/data/ets_grnch_attrs.json:77]: I'd vote to remove it yes, because it's misleads into thinking it should be easily accessible.

**Review by tlogan2000 — APPROVED** [review]
Looks ok to me.  Question is `xesmf` as a dependency a potential issue?  I don't think we end up needing `conservative_regrid` at least for now... If you see a use in keeping it I let you decide

**Pull request merged**